### PR TITLE
dnsmasq and firewall3 changes

### PR DESCRIPTION
**Pull request by ambrice**
Some changes needed to get dnsmasq running, and firewall3 closer to running.

firewall3 still doesn't run for me due to some missing iptables extensions.

### CODE REVIEW

File: recipes-tweaks/dnsmasq/dnsmasq_%.bbappend [hunk 1]
```diff
@@ -3,18 +3,28 @@ inherit openwrt
 FILESEXTRAPATHS_prepend := "${THISDIR}/${PN}:"
 
 SRC_URI += "git://github.com/openwrt/openwrt.git;name=openwrt;destsuffix=git/openwrt/;protocol=git;branch=chaos_calmer"
+SRC_URI += "file://99-dnsmasq.rules"
 
 SRCREV_openwrt = "${OPENWRT_SRCREV}"
 
+RDEPENDS_dnsmasq += "jsonpath"
```
Comment (kraj) [inline recipes-tweaks/dnsmasq/dnsmasq_%.bbappend]: this should be RDEPENDS_${PN}

File: recipes-core/xtables-addons/xtables-addons_2.12.bb [hunk 2]
```diff
@@ -0,0 +1,29 @@
+DESCRIPTION = "Xtables-addons contains a set of possibly useful but not included in the mainline kernel nefilter extensions"
+LICENSE = "GPLv2"
+LIC_FILES_CHKSUM = "file://LICENSE;md5=751419260aa954499f7abaabaa882bbe"
+DEPENDS = "virtual/kernel iptables"
+RDEPENDS_${PN} += "perl"
+
+inherit autotools module-base
+
+INC_PR = "r0"
```
Comment (kraj) [inline recipes-core/xtables-addons/xtables-addons_2.12.bb]: Drop this

File: recipes-core/xtables-addons/xtables-addons_2.12.bb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        file://200-add-lua-packetscript.patch \
+        file://201-fix-lua-packetscript.patch \
+        file://202-add-lua-autoconf.patch \
+        file://300-geoip-endian-detection.patch \
+        "
+
+SRC_URI[md5sum] = "aed5ce0873709ac243f1177fc81ff452"
+SRC_URI[sha256sum] = "c4865aa1c64c5ff173ff7b5d69425466c71f0f9b5eb5299c52c68bdcd46fa63b"
+
+EXTRA_OECONF = "--with-kbuild=${STAGING_KERNEL_DIR} --with-xtlibdir=/usr/lib/iptables"
```
Comment (kraj) [inline recipes-core/xtables-addons/xtables-addons_2.12.bb]: use ${libdir} instead of /usr/lib

File: recipes-core/ipset/ipset_6.24.bb [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+inherit autotools openwrt
+
+SRC_URI = "http://ipset.netfilter.org/${PN}-${PV}.tar.bz2"
+
+SRC_URI[md5sum] = "8831b8f01458bf2abacc222884195a62"
+SRC_URI[sha256sum] = "3071fc283f00a6472b5b352ef57f9825c9face70dda5b0d8715f8d43d0e995d0"
+
+DEPENDS += "libmnl"
+
+FILES_${PN}-dev += "${libdir}/*.so"
```
Comment (kraj) [inline recipes-core/ipset/ipset_6.24.bb]: this seems redundant

File: recipes-core/xtables-addons/xtables-addons_2.12.bb (hunk 2 above)
Comment (ambrice) [inline recipes-core/xtables-addons/xtables-addons_2.12.bb]: By "this" you mean the INC_PR?  I don't think it'll build without the inherit..

File: recipes-core/packagegroups/packagegroup-core-boot-openwrt.bb [hunk 5]
```diff
@@ -32,6 +32,7 @@ RDEPENDS_${PN} = "\
     base-passwd \
     busybox \
     bridge-utils \
+    tcpdump \
```
Comment (mattsm) [inline recipes-core/packagegroups/packagegroup-core-boot-openwrt.bb]: Should this be in a debug package group?

File: recipes-core/xtables-addons/xtables-addons_2.12.bb (hunk 2 above)
Comment (kraj) [inline recipes-core/xtables-addons/xtables-addons_2.12.bb]: PR is automatic now a days we dont need to set up INC_PR

File: recipes-core/packagegroups/packagegroup-core-boot-openwrt.bb (hunk 5 above)
Comment (kraj) [inline recipes-core/packagegroups/packagegroup-core-boot-openwrt.bb]: thats a good point.

Comment (ambrice) [inline recipes-core/packagegroups/packagegroup-core-boot-openwrt.bb]: I can remove it from core-boot if you like but I'm not sure how to create a debug package group and where to add it in.

File: recipes-core/ipset/ipset_6.24.bb (hunk 4 above)
Comment (ambrice) [inline recipes-core/ipset/ipset_6.24.bb]: The reason why this was needed was this line in openwrt.bbclass:

    FILES_${PN}  += "${libdir}/* ${datadir}/lua/5.*/"

Do you remember why that was needed?

**Comment by kraj** [comment]
On 17-03-10 16:29:32, Aaron Brice wrote:
> ambrice commented on this pull request.
> 
> 
> 
> > +DESCRIPTION = "Administration tool for IP sets"
> +HOMEPAGE = "http://ipset.netfilter.org"
> +LICENSE = "GPL-2.0"
> +LIC_FILES_CHKSUM = "file://COPYING;md5=59530bdf33659b29e73d4adb9f9f6552"
> +SECTION = "base"
> +
> +inherit autotools openwrt
> +
> +SRC_URI = "http://ipset.netfilter.org/${PN}-${PV}.tar.bz2"
> +
> +SRC_URI[md5sum] = "8831b8f01458bf2abacc222884195a62"
> +SRC_URI[sha256sum] = "3071fc283f00a6472b5b352ef57f9825c9face70dda5b0d8715f8d43d0e995d0"
> +
> +DEPENDS += "libmnl"
> +
> +FILES_${PN}-dev += "${libdir}/*.so"
> 
> The reason why this was needed was this line in openwrt.bbclass:
> 
>     FILES_${PN}  += "${libdir}/* ${datadir}/lua/5.*/"
> 
> Do you remember why that was needed?

I think there were some packages producing unversioning shared libraries

File: recipes-core/ipset/ipset_6.24.bb (hunk 4 above)
Comment (ambrice) [inline recipes-core/ipset/ipset_6.24.bb]: Also above that is:

    FILES_SOLIBSDEV = ""

I'm guessing that was a workaround for the fact that openwrt stuff doesn't use versioned libraries.  Come to think of it, the ipset recipe shouldn't inherit openwrt anyway..

Comment (kraj) [inline recipes-core/ipset/ipset_6.24.bb]: precisely yes.

File: recipes-core/packagegroups/packagegroup-core-boot-openwrt.bb (hunk 5 above)
Comment (mattsm) [inline recipes-core/packagegroups/packagegroup-core-boot-openwrt.bb]: You can probably leave it for now, but at some point a good distro / image debug package set would be nice. tcpdump, strace, perf, etc. Network related items. I'd have to look at other examples to see exactly how to do it though. Khem probably knows right away of a decent example.

**Comment by kraj** [comment]
since I felt the reviews are minor, we can address them with a followup.

**Comment by ambrice** [comment]
Thanks.  I got stuck on removing the openwrt class from the ipset recipe, for some reason it stopped building for me.  Didn't have time to work on it this week..

**Pull request merged**